src/sentry/workflow_engine/models/data_condition_group.py

```diff
@@ -10,12 +13,22 @@ class DataConditionGroup(DefaultFieldsModel):
     A data group is a way to specify a group of conditions that must be met for a workflow action to execute
     """
 
+    objects: ClassVar[BaseManager[Self]] = BaseManager(cache_fields=["id"])
+
     __relocation_scope__ = RelocationScope.Organization
     __repr__ = sane_repr("logic_type")
 
     class Type(models.TextChoices):
+        # ANY will evaluate all conditions, and return true if any of those are met
         ANY = "any"
+
+        # ANY_SHORT_CIRCUIT will stop evaluating conditions as soon as one is met
+        ANY_SHORT_CIRCUIT = "any-short"
```

Comment:
Is there a case where we _wouldn't_ want this behavior?

Reply:
yeah - one example is on a [detector's conditions](https://github.com/getsentry/sentry/pull/80513/files#diff-5be652c4f9cfde75ca59b124eefeb643a1147a011cea3cd328b89e27b8e514c1R280). 

This is needed for two reasons; 1 because the conditions aren't ordered, so we want to make sure to get the highest condition. and 2. because we may want to grab _all_ the actions that could have been triggered in this state changes (ok -> low -> high), dedupe the actions, and finally trigger them.